src/misc.rs

```diff
@@ -0,0 +1,11 @@
+use esp_idf_sys::*;
+
+pub fn get_default_efuse_mac() -> Result<[u8; 6], EspError> {
```

Comment:
I don't feel very comfortable wrapping this function, unless there are multiple - as in many - many use-cases where we need it.
If you are just going to use it in  e.g. the smtp implementation and a couple of others - then I rather say - we pass on it.

Reply:
This, like the nvs api is mostly to have safe wrappers around esp idf functions, so I don't have to have unsafe code in my project

I suppose if this is mostly meant as an implementation of `embedded-svc` there could be another crate just for safe wrappers (which this crate could make use of)

Reply:
> This, like the nvs api is mostly to have safe wrappers around esp idf functions, so I don't have to have unsafe code in my project
> 
> I suppose if this is mostly meant as an implementation of `embedded-svc` there could be another crate just for safe wrappers (which this crate could make use of)

No it is not necessarily just an implementation of `embedded-svc`. Take a look at `EspNetif`. This is currently not having a corresponding `embedded-svc` trait, as I'm not sure that the notion of a network interface can be easily exposed everywhere (as in on all platforms) as a top level notion.

But `esp-idf-svc` IS about exposing *self-contained* pieces of functionality, rather than individual functions on "as needed" basis, because the moment you expose an individual function "because I just need it for this one use case", you have to maintain and evolve it, which is not easy and results in breaking changes if you don't think upfront how this is going to evolve. If I have to give you example precisely about the `get_default_efuse_mac` function you've exposed: this is just one API call from a whole set of APIs: `esp_mac.h`. So, how do we think about the whole set of APIs which are out there in `esp_mac.h`?
* Are we expecting to expose all of these over time? If yes, would all of these be in `misc`, or do we need to dedicate a separate namespace (a.k.a. a Rust module) for them, rather than keeping them in `misc`? if so, how about we dedicate a module for all of these, from the get-go?
* if we don't expect that this whole API would be necessary, and we'll only need `get_default_efuse_mac` - for whatever reasons - at one place, perhaps it is not worth it exposing it as a public, safe API in the first place, as it won't be having that many usages anyway?

Reply:
Hmm, yes I understand your concerns

The reason I exposed it was to generate a unique Wi-Fi name (which seems to be a common use case)
But exposing it like this is probably too low level

Reply:
What do you mean by `unique Wi-Fi name`? A MAC (as in OSI layer 2 ethernet MAC)?
Huh. I was expecting that this is somehow handled by the ESP WiFi driver itself. Now, if it is not, we should fix the `EspWifi` code itself to derive a useful MAC or something. Ditto for `EspEth`.

But, if you don't need this MAC in the captive DNS portal, or in the SNTP implementation (and I'm curious why would you need a layer 2 thing in  layer 3 protocols anyway), then I say let's open a new issue describing the problem, and then fix it in the `EspWifi` and `EspEth` driver wrappers themselves.